npc/jobs/valkyrie.txt

```diff
@@ -217,45 +217,165 @@ yuno_in02,88,164,5	script	Metheus Sylphe#Library	2_M_SAGE_B,{
+						mes "[Metheus Sylphe]";
+						mes "I can feel the passion within you, but I couldn't let you enter for free.";
+						next;
+						mes "[Metheus Sylphe]";
+						mes "So help me resolve the problems happen in the Schweicherbil Magic Academy's library.";
+						mes "I'll help you to read the book of Ymir.";
+						do {
+							next;
+							switch(select("What happened?", "Where does it escaped?", "I'll find it.", "Ignore")) {
+								case 1:
+									mes "[Metheus Sylphe]";
+									mes "This is really a headache.";
+									mes "The library consists of a lot undisclosed books.";
+									next;
+									mes "[Metheus Sylphe]";
+									mes "For example, ^0055FFRideword Hat^000000 that would attack humans can't be disclose to the public.";
+									next;
+									mes "[Metheus Sylphe]";
+									mes "But not long ago, part of the undisclosed books escaped from the library...";
+									next;
+									mes "[Metheus Sylphe]";
+									mes "Although the library guards are currently searching for it, but failed due to lack of manpower.";
+									break;
+								case 2:
+									mes "[Metheus Sylphe]";
+									mes "We're still trying to figure out where it escaped to.";
+									mes "I'll inform you when the library guards have any news for it.";
+									break;
+								case 3:
+									mes "[Metheus Sylphe]";
+									mes "Do you understand the problem now?";
+									mes "Or you know other problems?";
+									next;
+									if (select("Sorry.", "Worry about gifts for girlfriend?", "Worry about the lunch?", "Worry about the escaped books?") == 4) {
+										mes "[Metheus Sylphe]";
+										mes "Ah! You knew the problem already?";
+										mes "or the news already leaked to public?";
+										next;
+										mes "[Metheus Sylphe]";
+										mes "Wait a moment, the library guard has returned with the report.";
+										next;
+										mes "[Metheus Sylphe]";
+										mes "Let's hear the report before you left!";
+										if (!questprogress(5160)) 
+											setquest 5160; // Quest #5160 - Library Mistake
+									}
+								default: 
+									close;
+							}
+						} while (true);
+					}
+					close;
+				default:
+					break;
 			}
 			mes "[Metheus Sylphe]";
 			mes "Take your time, and";
 			mes "enjoy your travels.";
 			close;
 		}
 		mes "[Metheus Sylphe]";
-		mes "Once again, thank you for your generous donation. Feel free to read a carbon copy of the 'Book of Ymir' at your leisure.";
+		mes "Once again, thank you for your generous "+(questprogress(5160) ? "assistance":"donation")+". Feel free to read a carbon copy of the 'Book of Ymir' at your leisure.";
 		close;
 	}
 	mes "[Metheus Sylphe]";
 	mes "Welcome to the Library of the Schweicherbil Magic Academy.";
 	mes "Here, we have a countless number of books. Please take your time and look around.";
 	close;
+	
+	OnKill:
+		if (questprogress(5160)) { // Quest #5160 - Library Mistake
+			if (!countitem(6586))
```

Comment:
This is incorrect, you can get multiple books and they will all get deleted by the npc.

Reply:
updated